_posts/2020/2020-05-15-constraints-and-freedom-writing-perfect-program.md

```diff
@@ -38,16 +37,17 @@ public class Sum {
 Read two integers from the input, calculate their sum and print the result to the output.
 What could be easier?
 
-## First obstacle
+## First obstacles
 As soon as we wrote the code and ran it on our computers, we realized that there are some things that could be improved.
-What if the user accidentally adds a non-digit character (such as a letter) to his number?
+What if the user accidentally adds a non-digit character (such as a letter) to their number?
 That would immediately raise the `InputMismatchException` and our contender for the "*Easiest program since Hello World*" title would break in a blink of an eye.
 Not something a perfect program should do.
 
-Obvious solution is to not allow any characters other than the numbers.
-We can't, however, proceed with that, as it will limit the user to the digit characters only.
-What about the real numbers with decimal points, such as the [gravity of Earth](https://en.wikipedia.org/wiki/Gravity_of_Earth){:target="_blank"} that equals to`9.8`? This number would already break the code above, by the way, as it's not an integer and Java's `Scanner` object did not expect it.
-Regardless, `9.8` is still a valid number, and we have never stated that our perfect program only works with integers, but rather with numbers in general, so we cannot let that *bring us down* (pun intended).
+The obvious solution was to not allow any characters other than numbers.
+We couldn't, however, proceed with that, as it will limit the user to the digit characters only.
```

Comment:
My intention here was to have a conversation with a reader, as _both of us_ are trying to find the perfect program together